go/service/chat_local.go

```diff
@@ -219,6 +219,9 @@ func (h *chatLocalHandler) GetThreadLocal(ctx context.Context, arg chat1.GetThre
 		return chat1.GetThreadLocalRes{}, err
 	}
 
+	// Run type filter if it exists
+	thread.Messages = chat.FilterByType(thread.Messages, arg.Query)
```

Comment:
I wanted it to work on the same type as what gets pulled from the local storage cache.